lib/middleware/bearerToken.ts

```diff
@@ -0,0 +1,60 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import jwt from "jsonwebtoken";
+import dbConnector from "@lib/integration/dbConnector";
+import executeQuery from "@lib/integration/queryManager";
+import { QueryResult } from "pg";
+
+export interface BearerTokenPayload {
+  formID?: string;
+}
+
+const validate = (
+  handler: (req: NextApiRequest, res: NextApiResponse, options?: unknown) => void
+) => {
+  return async (req: NextApiRequest, res: NextApiResponse): Promise<unknown> => {
+    try {
+      const token = getBearerToken(req);
+      const bearerTokenPayload = jwt.verify(token, process.env.TOKEN_SECRET || "");
```

Comment:
Bearer token also needs to be validated to be the same value in the database. If a new bearer token is generated and you do not check this then there is no way to invalidate old tokens

Reply:
Updated `bearerToken.ts` with a function to check the database.